make.js

```diff
@@ -0,0 +1,22 @@
+let make = (n) => {
+  let fn = function(x) {
+
+    let res = 0;
+
+    if (typeof x !== 'function') {
+      res = Array.from(arguments).reduce((accumulator, currentValue) => accumulator + currentValue);
+    }
+    return make(res+n);
+
+  }
+
+  fn.valueOf = () => n;
+
+  return fn;
+}
+
+function sum(a, b) {
+  return a + b;
+}
+
+module.exports = make;
```

Comment:
As for me, i would like to see here more expanded version without valueOf method.  But, anyway, thank you for your creativity.